src/Residue.php

```diff
@@ -69,120 +143,54 @@ public function decimal(int $decimal): self
 
         $this->decimal = $decimal;
 
+        $this->step = 10 ** -$decimal;
```

Comment:
I've a strange behavior following the order of the method (step/decimal). 

here a test to illustrate:

```php

    /**
     * @test
     */
    public function it_should_split_test_case(): void
    {
        // work as expected
        $residue = Residue::create(7.315)
                    ->divideBy(3)
                    ->decimal(3)
                    ->step(0.050);

        $this->assertEquals([2.45, 2.45, 2.4], $residue->toArray());
        $this->assertEquals(0.015, $residue->getRemainder());

        // fail
        $residue = Residue::create(7.315)
                    ->divideBy(3)
                    ->step(0.050) // <-- step before decimal
                    ->decimal(3);

        $this->assertEquals([2.45, 2.45, 2.4], $residue->toArray());
        $this->assertEquals(0.015, $residue->getRemainder());
        /*
             * Failed asserting that two arrays are equal.
            --- Expected
            +++ Actual
            @@ @@
            Array (
            -    0 => 2.45
            -    1 => 2.45
            -    2 => 2.4
            +    0 => 2.439
            +    1 => 2.438
            +    2 => 2.438
            )
         */
    }
```

maybe don't overwrite step if it was previously defined

Reply:
The trick here is that step is set to 0.05 so it doesn't make sense that when decimal is set to 3 the behavior is like that. Since 0.001 is smaller than 0.05 anyway so can't add any more precision...

step should actually only be rewritten if decimal is strictly smaller than step...

because if we do opposite :

```php
$residue = Residue::create(7.315)
                    ->divideBy(3)
                    ->step(0.050) // <-- step before decimal
                    ->decimal(1);

//One should expect a result of this type :

$this->assertEquals([2.5, 2.4, 2.4], $residue->toArray());
$this->assertEquals(0.015, $residue->getRemainder());
```

What do you think about it?

Reply:
> @romainnorberg maybe : don't overwrite step if it was previously defined
> @MockingMagician : step should actually only be rewritten if decimal is strictly greather than step...

I have pushed new implementation in that way that take care of both, you can know give it a try, if you find some edge cases to fix, it will be fine!

Reply:
yeah @MockingMagician it's work well 👍 Behavior is ok for me